### DISCUSSION TITLE
Setting custom width for modal

### DISCUSSION THREAD

**Discussion by OvalMedia** [filamentphp/filament#5468] [Help] [answered]
There is an undocumented method `modalWidth()` for actions. By default a modal is rather narrow as it usually prompts a simple dialog.
But how to set a custom width for a modal when I need a more complex form? Whatever parameter I pass along the modal always ends up being 100% wide.

**Comment by danharrin** [comment, marked as answer]
The options are based on Tailwind's spacing scale - https://tailwindcss.com/docs/max-width

**Reply by OvalMedia** [reply]
No matter which of the listed values I set it always ends up the same, the modal stretches to the complete width of the screen.
E.g. I tried this: `modalWidth('max-w-lg')`

**Reply by danharrin** [reply]
`modalWidth('lg')`

**Reply by divdax** [reply]
Hey Dan, i have commented out the create/edit pages. How i can set the create modal width?

**Reply by danharrin** [reply]
`CreateAction::make()->modalWidth(...)` etc

**Reply by madushancs** [reply]
How can I set the Relation Manager create/edit modal width?

**Reply by madushancs** [reply]
![image](https://github.com/filamentphp/filament/assets/45415384/6ccb5561-0050-4ca0-951a-147f1dd92af3)

**Reply by leandrocfe** [reply]
```php
$table
    ->headerActions([
        Tables\Actions\CreateAction::make()
            ->modalWidth('7xl')
...
```

**Reply by madushancs** [reply]
Thank you very much. Can we move next button to create button place?
![image](https://github.com/filamentphp/filament/assets/45415384/b1e0df38-e4dc-448b-9d47-9cb0089627b0)

**Reply by leandrocfe** [reply]
You should use `->steps([])` if you are using an action
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#using-a-wizard
